### PR TITLE
refactor: deprecate TEdgeConfigRepository

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

Deprecate `struct TEdgeConfigRepository` which was useless wrapper around `TEdgeConfigLocation`.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2153

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge/src/cli/reconnect/command.rs [hunk 1]
```diff
@@ -55,7 +53,10 @@ impl From<&ReconnectBridgeCommand> for ConnectCommand {
     fn from(reconnect_cmd: &ReconnectBridgeCommand) -> Self {
         ConnectCommand {
             config_location: reconnect_cmd.config_location.clone(),
-            config_repository: reconnect_cmd.config_repository.clone(),
+            // FIXME: cloning the config leads to the following error:
+            //        expected `TEdgeConfig`, found `TEdgeConfigReader`
+            // config: reconnect_cmd.config.clone(),
+            config: TEdgeConfig::new(reconnect_cmd.config_location.clone()).unwrap(),
```
Comment (didier-wenzek) [inline crates/core/tedge/src/cli/reconnect/command.rs:59]: @jarhodes-sag any ideas on why this error and how this can be fixed?

Comment (didier-wenzek) [reply crates/core/tedge/src/cli/reconnect/command.rs]: Fixed by https://github.com/thin-edge/thin-edge.io/pull/2735/commits/e5eec7519f5315fd0a4123236834bd7ddcacbdb3

File: crates/common/tedge_config/src/lib.rs [hunk 2]
```diff
@@ -7,21 +7,34 @@ pub use self::tedge_config_cli::error::*;
...
 pub use tedge_config_macros::all_or_nothing;
 pub use tedge_config_macros::OptionalConfig;
 
-/// loads the new tedge config from system default
-pub fn get_new_tedge_config() -> Result<TEdgeConfig, TEdgeConfigError> {
-    let tedge_config_location = TEdgeConfigLocation::default();
-    TEdgeConfigRepository::new(tedge_config_location).load()
-}
+impl TEdgeConfig {
+    pub fn new(config_location: TEdgeConfigLocation) -> Result<Self, TEdgeConfigError> {
```
Comment (albinsuresh) [inline crates/common/tedge_config/src/lib.rs:17]: Minor(idiomatic):
```suggestion
    pub fn try_new(config_location: TEdgeConfigLocation) -> Result<Self, TEdgeConfigError> {
```

File: crates/common/tedge_config/src/lib.rs [hunk 3]
```diff
@@ -7,21 +7,34 @@ pub use self::tedge_config_cli::error::*;
...
+        config_location.load()
+    }
+
+    pub fn load(config_dir: &Path) -> Result<TEdgeConfig, TEdgeConfigError> {
```
Comment (albinsuresh) [inline crates/common/tedge_config/src/lib.rs:21]: Wondering if we could just merge the `new` and this `load` functions into a single one that takes an `AsRef<Path>` as the argument, with an `impl AsRef<Path> for TEdgeConfigLocation`.

**Review by albinsuresh — APPROVED** [review]

File: crates/common/tedge_config/src/lib.rs (hunk 3 above)
Comment (didier-wenzek) [reply crates/common/tedge_config/src/lib.rs:21]: This was my initial plan. Unfortunately, this is more involved as in many places both the config and its location are used. We have first to simplify the configuration of misc actors and plugins (notably in the c8y mapper).

File: crates/common/tedge_config/src/lib.rs (hunk 2 above)
Comment (didier-wenzek) [reply crates/common/tedge_config/src/lib.rs]: See https://github.com/thin-edge/thin-edge.io/pull/2735/commits/cb96c2d2528174ee83d8a18f803be1fcddcdd3bf

**Pull request merged**